plutus-ledger-api/src/Plutus/V1/Ledger/Interval.hs

```diff
@@ -260,11 +260,17 @@ contains (Interval l1 h1) (Interval l2 h2) = l1 <= l2 && h2 <= h1
 
 {-# INLINABLE isEmpty #-}
 -- | Check if an 'Interval' is empty.
-isEmpty :: Ord a => Interval a -> Bool
+isEmpty :: (Enum a, Ord a) => Interval a -> Bool
 isEmpty (Interval (LowerBound v1 in1) (UpperBound v2 in2)) = case v1 `compare` v2 of
-    LT -> False
+    LT -> if openInterval then checkEnds v1 v2 else False
     GT -> True
     EQ -> not (in1 && in2)
+    where
+        openInterval = in1 == False && in2 == False
+        -- | We check two finite ends to figure out if there are elements between them.
+        -- If there are no elements then the interval is empty (#3467).
+        checkEnds (Finite v1') (Finite v2') = (succ v1') `compare` v2' == EQ
```

Comment:
I had to think about this a bit. I wonder whether it would be clearer to express it like this:
```
toClosedInterval (Enum a) => Interval a -> Bool
toClosedInterval (Interval l@(LowerBound v1 in1) u@(UpperBound v2 in1)) = 
    Interval (if not in1 then (LowerBound (succ v1) True))) (if not in2 then (UpperBound (pred v2) True) else u)

-- just putting a view pattern here to make it as similar as possible to before
isEmpty (toClosedInterval -> (Interval ...)) = case v1 `compare` v2 of
  LT -> False
  ...
```

That is, what `Enum` gives you is the ability to turn an open endpoints into equivalent closed endpoints, and then we simply only check emptiness on closed intervals.

Reply:
Up to you, though, my version is more complex.

Reply:
I agree, your version looks more general and more clear in some sense.

`succ v1` requires `Enum` instance for `Extended a` though... I avoided to write it because less code is better :D